internal/service/grid/extensibleattributedef_resource.go

```diff
@@ -0,0 +1,218 @@
+			fmt.Sprintf("Expected *niosclient.APIClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *ExtensibleattributedefResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data ExtensibleattributedefModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiRes, _, err := r.client.GridAPI.
+		ExtensibleattributedefAPI.
+		Create(ctx).
+		Extensibleattributedef(*data.Expand(ctx, &resp.Diagnostics, true)).
+		ReturnFieldsPlus(readableAttributesForExtensibleattributedef).
+		ReturnAsObject(1).
+		Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create Extensibleattributedef, got error: %s", err))
+		return
+	}
+
+	res := apiRes.CreateExtensibleattributedefResponseAsObject.GetResult()
+
+	data.Flatten(ctx, &res, &resp.Diagnostics)
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ExtensibleattributedefResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data ExtensibleattributedefModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiRes, httpRes, err := r.client.GridAPI.
+		ExtensibleattributedefAPI.
+		Read(ctx, utils.ExtractResourceRef(data.Ref.ValueString())).
+		ReturnFieldsPlus(readableAttributesForExtensibleattributedef).
+		ReturnAsObject(1).
+		Execute()
+
+	// Handle not found case
+	if err != nil {
+		if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
+			// Resource no longer exists, remove from state
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read Extensibleattributedef, got error: %s", err))
+		return
+	}
+
+	res := apiRes.GetExtensibleattributedefResponseObjectAsResult.GetResult()
+
+	data.Flatten(ctx, &res, &resp.Diagnostics)
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ExtensibleattributedefResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var diags diag.Diagnostics
+	var data ExtensibleattributedefModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	diags = req.State.GetAttribute(ctx, path.Root("ref"), &data.Ref)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	apiRes, _, err := r.client.GridAPI.
+		ExtensibleattributedefAPI.
+		Update(ctx, utils.ExtractResourceRef(data.Ref.ValueString())).
+		Extensibleattributedef(*data.Expand(ctx, &resp.Diagnostics, true)).
```

Comment:
```suggestion
		Extensibleattributedef(*data.Expand(ctx, &resp.Diagnostics, false)).
```

For Update , IsCreate Should be set to false.